docs: add warning about dynamic import of css

InerkyJad opened the pull request:
The warning is regarding the use of dynamic imports inside of the script tag, it's related to #21803

<!---
☝️ PR title should follow conventional commits (https://conventionalcommits.org)

Please carefully read the contribution docs before creating a pull request
 👉 https://nuxt.com/docs/community/contribution
-->

### 🔗 Linked issue

<!-- Please ensure there is an open issue and mention its number as #123 -->

### ❓ Type of change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

<!-- Describe your changes in detail -->
<!-- Why is this change required? What problem does it solve? -->
<!-- If it resolves an open issue, please link to the issue here. For example "Resolves #1337" -->

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

InerkyJad committed:
Update 4.styling.md : added an alert

The warning is regarding the use of dynamic imports inside of the script tag, it's related to #21803

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

MuhammadM1998 commented:
Hello,
Maybe you can add an example of a static import ( `import "./style.css`) too as the current line maybe confusing.

Is this seems better?
```
Make sure to use a **synchronous** import `import "./style.css` instead of a **dynamic** import `import("~/assets/style.css")` for server-side compatibility.
```

florian-lefebvre reviewed (changes requested):
I agree with @MuhammadM1998, adding an example of synchronous import is less confusing. Could you add it?

florian-lefebvre commented:
@Hebilicious can you review?

danielroe commented:
Maybe add this as a comment to the existing example of synchronous import above?

InerkyJad committed:
Merge branch 'main' into patch-1

InerkyJad committed:
docs: comments on the style import in 4.styling.md 

Using comments to showcase both examples and warning against the use of dynamic imports.

InerkyJad committed:
docs: Update removal of old Alert in 4.styling.md

danielroe reviewed (approved):
Thank you - perfect!

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

florian-lefebvre commented on docs/1.getting-started/4.styling.md:
```diff
@@ -18,7 +18,11 @@ You can use a javascript import, or a css [`@import` statement](https://develope
+// Use a static import for server-side compatibility
 import '~/assets/css/first.css'
+
+// Caution: Dynamic imports are not server-side compatible.
```
```suggestion
// Caution: Dynamic imports are not server-side compatible
```
To keep it consistent with the previous comment

danielroe committed:
chore: remove full stop

Co-authored-by: Florian Lefebvre <contact@florian-lefebvre.dev>
